feat(Geotastic): add presence

AzNate opened the pull request:
## Description 
A new presence for the geography location guessing game, Geotastic. Detects all pages and displays info for all game modes.

## Acknowledgements
✓ I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
✓ I linted the code by running `yarn format`

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

![image](https://user-images.githubusercontent.com/47200527/170112606-aad1b45a-c115-4c08-80e2-71e8a5786101.png)

![image](https://user-images.githubusercontent.com/47200527/170112718-91eba2cf-20c1-4d21-9cdf-d9860fe76293.png)
</details>

AzNate committed:
New presence for Geotastic

Presence for the geography location guessing game, Geotastic.

AzNate committed:
Update presence.ts

Replaced var with let

AzNate committed:
Update presence.ts

Addressing Lint issues

EncryptedDev commented on websites/G/Geotastic/presence.ts:
```diff
@@ -0,0 +1,150 @@
+									case "/account/account-settings":
+										{
+											presenceData.details = "Viewing their account settings";
+											// No default
```
```suggestion
```

EncryptedDev commented:
Awaiting @Kozou4ever approval

Kozou4ever commented on websites/G/Geotastic/dist/metadata.json:
```diff
@@ -0,0 +1,23 @@
+	"service": "Geotastic",
+	"description": {
+		"en": "Geotastic is a geographical location guessing game created by Eduard But from Edutastic Games.",
+		"fr": "Geotastic est un jeu de devinette de emplacement géographique créé par Eduard But de Edutastic Games."
```
This translation would be the most appropriate in my opinion 
```suggestion
		"fr": "Geotastic est un jeu de devinette de localisation géographique créé par Eduard But de Edutastic Games."
```

AzNate committed:
Update websites/G/Geotastic/dist/metadata.json

Co-authored-by: Kozou <72155852+Kozou4ever@users.noreply.github.com>

ImRodry commented on websites/G/Geotastic/presence.ts:
```diff
@@ -0,0 +1,149 @@
+			}`;
+		} else presenceData.details = "Playing a game";
+	} else {
+		switch (document.location.pathname) {
```
why can't this start at the top?

ImRodry commented on websites/G/Geotastic/presence.ts:
```diff
@@ -0,0 +1,149 @@
+				else if (document.location.pathname.includes("/online-lobby/"))
+					presenceData.details = "In an online lobby";
+				else {
+					switch (document.location.pathname) {
```
a switch inside a switch for the same thing? Come on

ImRodry commented on websites/G/Geotastic/presence.ts:
```diff
@@ -0,0 +1,149 @@
+							} else if (document.location.pathname.includes("/help-out/"))
+								presenceData.details = "Viewing ways to support Geotastic";
+							else {
+								switch (document.location.pathname) {
```
again... let's not do this please

ImRodry commented on websites/G/Geotastic/presence.ts:
```diff
@@ -0,0 +1,149 @@
+										presenceData.details = "Viewing their quota consumption";
+										break;
+									}
+									case "/account/account-settings":
+										{
+											presenceData.details = "Viewing their account settings";
+										}
+										break;
```
why did this one decide to be the quirky young sibling

AzNate committed:
Update presence.ts

Moved switch to top and removed redundant switches. Fixed issue with account settings.

AzNate committed:
Update presence.ts

Fix formatting issue

EncryptedDev commented on websites/G/Geotastic/presence.ts:
```diff
@@ -0,0 +1,142 @@
+		case "/map-editor/maps": {
+			presenceData.details = "Creating a map";
+			break;
+		}
+		case "/map-editor/drop-groups": {
+			presenceData.details = "Viewing their drop groups";
+			break;
+		}
+		case "/events": {
+			presenceData.details = "Browsing past and future events";
+			break;
+		}
+		case "/account/profile": {
+			presenceData.details = "Viewing their profile settings";
+			break;
+		}
+		case "/account/statistics": {
+			presenceData.details = "Viewing their statistics";
+			break;
+		}
+		case "/account/trophies": {
+			presenceData.details = "Viewing their trophies";
+			break;
+		}
+		case "/account/friends": {
+			presenceData.details = "Viewing their friends";
+			break;
+		}
+		case "/account/supporter-status": {
+			presenceData.details = "Viewing their supporter status";
+			break;
+		}
+		case "/account/quota-consumption": {
+			presenceData.details = "Viewing their quota consumption";
+			break;
+		}
+		case "/account/account-settings": {
+			presenceData.details = "Viewing their account settings";
+			break;
+		}
```
Please use a Record for this instead as shown below. Much shorter and more maintainable 
https://github.com/PreMiD/Presences/blob/113adabd2296ec1be22122ccc3ec2deff1c399ed/websites/T/TruckersMP/presence.ts#L15-L41

AzNate committed:
Update presence.ts

Replaced giant Switch with a Record

AzNate committed:
Update presence.ts

Addressing Lint issue

AzNate committed:
Update presence.ts

Fixed Lint issue

AzNate committed:
Update presence.ts

Addressed Lint issues again

AzNate committed:
Update presence.ts

Fix Lint issues

ImRodry committed:
style: fix formatting issues
